Improve the description of ub-risk-3 (#26)
google/rust-crate-audits: auditing_standards.md

Requires **Unsafe Rust expertise**

> Crates with this criteria contain unsafe Rust code which doesn't uphold the typical standards required for unsafe code. They MAY pose a risk of causing undefined behavior, and probably should be allowed on a case-by-case basis based on usage.

Reviewer: Should be "this criterion"

    *   Writing implementations of traits not marked `unsafe` by violating documented invariants.
    *   Writing implementations of traits not marked `unsafe` that are not really intended to be implemented by the user.
    *   Explicitly forgetting values that have important drop behavior to cause UB when combined with operations that would not be expected to follow normally.
*   Some caution may be required to avoid undefined behavior.

> `ub-risk-3` is expected to live at the threshold of acceptance as a dependency for most codebases: there is unsoundness in the crate, but there are also many situations where one need not worry about that unsoundness ever being triggered, and the codebase needs to make a judgement call. For example, one may choose to depend on a `ub-risk-3` dep if the rating is due to:

Reviewer 2: ```suggestion
`ub-risk-3` is the highest risk level at which non-experts could reasonably avoid causing undefined behavior. These crates are unsound, but the risk they pose may be acceptable in some situations. Projects need to make judgement calls about where, when, and by whom these crates may be used. For example, a project may deem a `ub-risk-3` crate acceptable to use if it contains unsoundness that:
```
Do you think that the lemma of "non-experts could reasonably avoid causing UB" is a useful one for determining `ub-risk-3`?

>  * Unsoundness that is only triggered on platforms that the codebase does not care about.

Reviewer 2: ```suggestion
 * Can only be triggered on unsupported targets.
```

Reviewer: I would say "targets the codebase doesn't use" or "unused targets". The targets are *supported*, in that the crate will presumably build for and run on them.

>  * Unsoundness that may become a problem in future versions of Rust, where the codebase is confident of having upstream patches by then.

Reviewer 2: ```suggestion
 * May only become a problem in future versions of Rust, at which time it will be confidently be patched.
```

>  * Unsoundness that may become a problem in future versions of Rust, where the codebase is confident of having upstream patches by then.

Reviewer: The codebase's developers are confident, not the codebase. :) How about something like:

* Unsoundness that might become a problem in future versions of Rust, and by which time the crate's callers can be reasonably confident that the crate will have been fixed.

(However, as a code reviewer of the import and/or of the audit CL(s), I'd be skeptical.)

>  * Unsoundness that is triggered by a particular pattern of use of an API, which the codebase does not care to do and is confident of being able to avoid long term.

Reviewer 2: ```suggestion
 * Is triggered by a pattern of use which the project discourages and can confidently avoid long-term.
```

> This is true for all audit levels, but `ub-risk-3` audits in particular should strive to contain audit comments detailing the safety issues found, to aid in such judgement calls.

Reviewer 2: ```suggestion
All audit levels should strive to detail the safety issues found. However, those details are especially important for `ub-risk-3` audits because projects depend on those details to determine whether high-risk crates are acceptable to use.
```

### `ub-risk-4`

Also called: "Extreme unsoundness", "Very risky"

Requires **Unsafe Rust expertise**

> Crates with this criteria contain very dangerous unsafe rust code. They pose a risk of causing undefined behavior with typical use.

Reviewer: "criterion"

Author: This is a broader grammar fix to the whole document that i won't do here
